semantic/deep_equal.go

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package semantic
+
+import (
+	"k8s.io/utils/third_party/forked/golang/reflect"
```

Comment:
are we spreading this pattern to other repos? is the `third_party/forked/golang` path segment required? should we make this part internal?

Reply:
third_party is required as per CNCF legal.  It could be under `internal/third_party`perhaps.

Reply:
done